src/AlertsManagement/AlertsManagement/ActionRuleCommands/UpdateAzureActionRule.cs

```diff
@@ -0,0 +1,150 @@
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Rest.Azure;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.AlertsManagement.OutputModels;
+using Microsoft.Azure.Management.AlertsManagement.Models;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.Commands.AlertsManagement
+{
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ActionRule", SupportsShouldProcess = true)]
+    [OutputType(typeof(PSActionRule))]
+    public class UpdateAzureActionRule : AlertsManagementBaseCmdlet
+    {
+        #region Parameter Set Names
+
+        private const string ByInputObjectParameterSet = "ByInputObject";
+        private const string ByResourceIdParameterSet = "ByResourceId";
+        private const string ByNameJsonPatchParameterSet = "ByNameJsonPatch";
+        private const string ByNameSimplifiedPatchParameterSet = "ByNameSimplifiedPatch";
+
+        #endregion
+
+        #region Parameters declarations
+        /// <summary>
+        /// Gets or sets the resource id parameter.
+        /// </summary>
+        [Parameter(ParameterSetName = ByResourceIdParameterSet, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource id of action rule")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the input object
+        /// </summary>
+        [Parameter(ParameterSetName = ByInputObjectParameterSet, Mandatory = true, ValueFromPipeline = true, HelpMessage = "The action rule resource")]
+        public PSActionRule InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the action rule name
+        /// </summary>
+        [Parameter(ParameterSetName = ByNameJsonPatchParameterSet, Mandatory = true, HelpMessage = "Action rule name")]
+        [Parameter(ParameterSetName = ByNameSimplifiedPatchParameterSet, Mandatory = true, HelpMessage = "Action rule name")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource group name
+        /// </summary>
+        [Parameter(ParameterSetName = ByNameJsonPatchParameterSet, Mandatory = true, HelpMessage = "Action rule name")]
+        [Parameter(ParameterSetName = ByNameSimplifiedPatchParameterSet, Mandatory = true, HelpMessage = "Action rule name")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets patch object
+        /// </summary>
+        [Parameter(ParameterSetName = ByNameJsonPatchParameterSet, Mandatory = true, HelpMessage = "Action rule patch object in JSON format")]
+        [ValidateNotNullOrEmpty]
+        public string ActionRulePatch { get; set; }
```

Comment:
It can patch - tags or changes status of action rule. It is broken down into parameters already. The Json input is another parameter set. It was my one of the queries that what is the recommended way to deal with complex parameters - is it break down, json string or both?